app/utils/utils.ts

```diff
@@ -98,7 +109,38 @@ export const apiCall = async (
   apiUrl += queryString;
 
   try {
-    const startTime = new Date().getTime();
+    const response = await RepoAPICall(apiUrl);
+
+    if (response.status === 200) {
+      const data = await response.json();
+      // console.log(`apiCall: called ${apiUrl}`);
+      // console.log(`Response time: ${new Date().getTime() - startTime}`);
+      return data.nyplAPI.response;
+    } else {
+      return undefined;
+    }
+  } catch (error) {
+    return undefined;
+  }
+};
+
+/**
+ * Returns Repo API response WITH request data.
```

Comment:
Why do you need to do this?

I'm not against this, but you can refactor these two functions so that the bulk of the work is not repeated. The base function can have `return data`. Then the other function can call the base function and return `data.nyplAPI.response`.

Reply:
one returns the whole api response and the other returns  `data.nyplAPI.response.` - wanted the request object from the original response to get the uuid and the pagination information.

Reply:
Right, that's ok. But you're doing the base work twice. I think you can keep `RepoAPICall` as is since that's where the `fetch` call is being made and let it return the base `data` response object. In `apiCall`, you don't need most of the code. You can do

```
export const apiCall = async (
  apiUrl: string,
  urlParam?: { [key: string]: any }
) => {
  const data = await RepoAPICall(apiUrl, urlParam);

  return data?.nyplAPI?.response;
};
```

- In your current `apiCall` function, you don't need to wrap `await RepoAPICall` in a try/catch or check for status because that's already done in the `RepoAPICall` since that function calls fetch. The error handling occurs in that function and if there is an error, it will return undefined. That's why in this function, you can just check for undefined values with the `?.` optional chaining.
- `RepoAPICall` returns the `data` object and `apiCall` returns the `response` property from the `nyplAPI` property. The names of the functions could be clearer.

Reply:
Yes, that makes sense.